tiledb/sm/c_api/tiledb.cc

```diff
@@ -1552,37 +1528,26 @@ capi_return_t tiledb_array_delete_fragments_list(
   }
 
   // Allocate an array object
-  tiledb_array_t* array = new (std::nothrow) tiledb_array_t;
-  try {
-    array->array_ =
-        make_shared<tiledb::sm::Array>(HERE(), ctx->resources(), uri);
-  } catch (...) {
-    delete array;
-    array = nullptr;
-    throw api::CAPIStatusException("Failed to create array");
-  }
+  tiledb::sm::Array array(ctx->resources(), uri);
 
   // Open the array for exclusive modification
-  throw_if_not_ok(array->array_->open(
-      static_cast<tiledb::sm::QueryType>(TILEDB_MODIFY_EXCLUSIVE),
-      static_cast<tiledb::sm::EncryptionType>(TILEDB_NO_ENCRYPTION),
+  throw_if_not_ok(array.open(
+      tiledb::sm::QueryType::MODIFY_EXCLUSIVE,
+      tiledb::sm::EncryptionType::NO_ENCRYPTION,
       nullptr,
       0));
 
   // Delete fragments list
   try {
-    array->array_->delete_fragments_list(uris);
+    array.delete_fragments_list(uris);
   } catch (...) {
-    throw_if_not_ok(array->array_->close());
-    delete array;
-    array = nullptr;
-    throw api::CAPIStatusException("Failed to delete fragments_list");
+    throw_if_not_ok(array.close());
+    std::throw_with_nested(
+        api::CAPIStatusException("Failed to delete fragments_list"));
```

Comment:
Ditto

Reply:
Reverted to what was before this PR because a test was depending on the exception's message.